### PR TITLE
fix(VBtn): isActive calc priorities

### LINKED ISSUE #15833: [Bug Report][3.0.0] Unable to make VBottomNavigation non-active on inactive route
### Environment
**Vuetify Version:** 3.0.0-beta.11
**Vue Version:** 3.2.39
**Browsers:** Chrome 105.0.0.0
**OS:** Windows 10

### Steps to reproduce
- Open reproduction in [stackblitz](https://stackblitz.com/edit/nuxt-starter-hmofvf?file=layouts/default.vue).
- Click on `/about` route (which is not available in VBottomNavigation).
- VBottomNavigation always show active button even on in-active route.

### Expected Behavior
All buttons in VBottomNavigation shall not be active on in-active route.

### Actual Behavior
VBottomNavigation always show active button even on in-active route

### Reproduction Link
<a href="https://stackblitz.com/edit/nuxt-starter-hmofvf?file=layouts/default.vue" target="_blank">https://stackblitz.com/edit/nuxt-starter-hmofvf?file=layouts/default.vue</a>

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #15833 & #15982

Cause: `v-bottom-navigation` adds `group?.isSelected`, which returns `isActive` as true even if VBtn's `link.isActive` becomes false

Solution: 

- `props.active` takes highest priority when it is explicitly set as `true` or `false`
- `link.active` takes second priority when it is a link
- `group.active` performs as the lowest priority when previous two are not valid

## Description

fixes #15833 & #15982

## How Has This Been Tested?

visually 

## Markup:

Firstly, config routes in `packages/vuetify/dev/index.js`

```js
// import { routes } from './router'
const Home = { template: '<div>Home</div>' }
const About = { template: '<div>About</div>' }
const Help = { template: '<div>Help</div>' }
const Login = { template: '<div>Login</div>' }

const routes = [
  { path: '/', component: Home },
  { path: '/about', component: About },
  { path: '/help', component: Help },
  { path: '/login', component: Login },
]
```

<details>

```vue
// Paste your FULL Playground.vue here
<template>
  <v-app>
    <v-main>
      <v-container>
        
        <v-card
          title="Unable to make VBottomNavigation non-active on inactive route"
          class="my-4"
        >
        <v-btn-toggle
        v-model="text"
        color="deep-purple accent-3"
      >
        <v-btn value="left" >
          Left
        </v-btn>

        <v-btn value="center" >
          Center
        </v-btn>

        <v-btn value="right" :active="false">
          Right
        </v-btn>

        <v-btn value="justify" :active="false">
          Justify
        </v-btn>
      </v-btn-toggle>
          <v-card-actions>
            <v-spacer />
            <v-btn
              v-for="{ path } of [
                { path: '/' },
                { path: '/help' },
                { path: '/about' },
              ]"
              color="primary"
              :active="true"
              variant="text"
              :to="path"
              class="text-none"
            >
              {{ path }}
            </v-btn>
          </v-card-actions>
        </v-card>
      </v-container>
    </v-main>

    <!-- How to make it non-active on /about? -->
    <v-bottom-navigation color="primary" :mandatory="false">
      <v-btn
        v-for="{ value, to, label } in [
          {
            value: 'home',
            to: '/',
            label: 'Home',
          },
          {
            value: 'help',
            to: '/help',
            label: 'Help',
          },
          {
            value: 'login',
            to: '/login',
            label: 'Login',
          },
        ]"
        :value="value"
        :to="to"
      >
        <span>{{ label }}</span>
      </v-btn>
    </v-bottom-navigation>
  </v-app>
</template>

<script setup lang="ts">
import { ref } from 'vue'

const text = ref()
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VBtn/VBtn.tsx [hunk 1]
```diff
@@ -110,7 +117,7 @@ export const VBtn = defineComponent({
 
     useRender(() => {
       const Tag = (link.isLink.value) ? 'a' : props.tag
-      const hasColor = !group || group.isSelected.value
+      const hasColor = !group || (isActive.value && group.isSelected.value)
```
Comment (johnleider) [inline packages/vuetify/src/components/VBtn/VBtn.tsx:120]:
```suggestion
      const hasColor = !group || isActive.value
```

Comment (johnleider) [reply packages/vuetify/src/components/VBtn/VBtn.tsx:120]: isActive already takes group.isSelected into account

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/components/VBtn/VBtn.tsx (hunk 1 above)
Comment (yuwu9145) [reply packages/vuetify/src/components/VBtn/VBtn.tsx]: Thanks! feedback is addressed

**Comment by yuwu9145** [comment]
@KaelWD I requested for a further review from John, but somehow it automatically removed you as reviewer. This was unintentional and feel free to check this out as well

**Review by johnleider — APPROVED** [review]

**Pull request merged**